lib/serializer/BinaryDeserializer.h

```diff
@@ -498,7 +498,7 @@ class DLL_LINKAGE BinaryDeserializer : public CLoaderBase
 		{
 			T t;
 			load(t);
-			data = std::move(t);
+			data = boost::make_optional(std::move(t));
```

Comment:
Tow suggested to use emplace here:
```
			data.emplace();
			load(*data);
```
It's seems to work well, but in other cases I not sure whatever we should use it since ```boost::make_optional``` seems to be easier to understand it's boost::optional value is assigned here.